### PR TITLE
feat: homework lesson #1

### PR DESCRIPTION
**Pull request by trongtai37**
## Done
* 136_single-number.ts      
* 191_number-1-bits.ts  
* 206_reverse_linked_list.ts  
* 304_range_sum_query_2d.ts  
* 905_sort_array_by_parity.ts
* 141_linked_list_cycle.ts 
* 202_happy_number.ts   
* 303_range_sum_query.ts      
* 78_subsets.ts
## Inprogress
N/A
## Todo
N/A

### CODE REVIEW

File: homework/tai_nguyen/141_linked_list_cycle.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+function detectCycle(head: ListNode | null): ListNode | null {
```
Comment (quanghoang33) [inline homework/tai_nguyen/141_linked_list_cycle.ts]: is this sol for https://leetcode.com/problems/linked-list-cycle-ii/ ?

File: homework/tai_nguyen/304_range_sum_query_2d.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  prefixSum2D: number[][];
+  constructor(matrix: number[][]) {
+    this.prefixSum2D = new Array(matrix.length)
+      .fill(true)
+      .map(() => new Array(matrix[0].length).fill(0));
+
+    matrix.forEach((row, i) => {
+      row.forEach((cell, j) => {
+        this.prefixSum2D[i][j] =
+          (this.prefixSum2D[i - 1]?.[j] ?? 0) +
```
Comment (quanghoang33) [inline homework/tai_nguyen/304_range_sum_query_2d.ts]: this should work but kind of hard to read and error-prone. Perhaps consider Thinh's [implementation](https://github.com/quanghoang33/gen-6/pull/4/files#diff-0098a80934b27d3c2ee8e2a5c818073c5b8bb4e1cbaec8e7d96f1bbaa756f86eR1) or Truong's [implementation](https://github.com/quanghoang33/gen-6/pull/2/files/6175d37c6c40c2b5661a57fd74da2eb456fc89d4#diff-e290a9c1157931476d8a9f8accea220a5eb2ab375c5656c51e6c4a50b207f763R1)

File: homework/tai_nguyen/141_linked_list_cycle.ts (hunk 1 above)
Comment (trongtai37) [reply homework/tai_nguyen/141_linked_list_cycle.ts]: Oh, thanks @quanghoang33. I miss it and commit code of another similar problem.
Let me update this file :)

File: homework/tai_nguyen/304_range_sum_query_2d.ts (hunk 2 above)
Comment (trongtai37) [reply homework/tai_nguyen/304_range_sum_query_2d.ts]: Yes, this line looks quite complex. I will fix it.
Thanks for your reference @quanghoang33

**Pull request merged**